src/Sharpliner/AzureDevOps/Model/Tasks/Marketplace/PublishTestResultsTask.cs

```diff
@@ -0,0 +1,143 @@
+﻿using Sharpliner.AzureDevOps.Tasks;
+using YamlDotNet.Serialization;
+
+namespace Sharpliner.AzureDevOps.Model.Tasks;
+
+/// <summary>
+/// More details can be found in <see href="https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/publish-test-results-v2">official Azure DevOps pipelines documentation</see>
+/// </summary>
+public record PublishTestResultsTask: AzureDevOpsTask
+{
+    public PublishTestResultsTask(TestResultsFormat testResultsFormat, string testResultsFile) : base("PublishTestResults@2")
+    {
+        TestResultsFormat = testResultsFormat;
+        TestResultsFiles = testResultsFile;
+    }
+
+    /// <summary>
+    /// Specifies the format of the results files you want to publish.
+    /// JUnit, NUnit, VSTest, XUnit, CTest
+    /// </summary>
+    [YamlIgnore]
+    public TestResultsFormat TestResultsFormat
+    {
+        get => GetEnum<TestResultsFormat>("testResultsFormat") ?? TestResultsFormat.JUnit;
+        init => SetProperty("testResultsFormat", value.ToString());
+    }
+
+    /// <summary>
+    /// Specifies one or more test results files.
+    /// You can use a single-folder wildcard (*) and recursive wildcards (**).
+    /// <example>**/TEST-*.xml searches for all the XML files whose names start with TEST- in all subdirectories.</example>
+    /// <remarks>If using VSTest as the test result format, the file type should be changed to .trx e.g. **/TEST-*.trx</remarks>
+    /// </summary>
+    [YamlIgnore]
+    public string TestResultsFiles
+    {
+        get => GetString("testResultsFiles") ?? "**/TEST-*.xml";
+        init => SetProperty("testResultsFiles", value);
+    }
+
+    /// <summary>
+    /// Specifies the folder to search for the test result files.
+    /// </summary>
+    [YamlIgnore]
+    public string SearchFolder
+    {
+        get => GetString("searchFolder") ?? "$(System.DefaultWorkingDirectory)";
```

Comment:
Same - why do we have a default? Shouldn't we return `null`?

Reply:
That's the default value when the input is not set as per docs

```
    #searchFolder: '$(System.DefaultWorkingDirectory)' # string. Search folder. Default: $(System.DefaultWorkingDirectory).
```

I added it as a default so that if the property isn't set, the following `get`s return the value ADO considers as default.

Do you propose that I should get rid of the default altogether?

Reply:
Removed defaults